### PR TITLE
feat(gatsby): Remove deleteNode with ID arg

### PR DESCRIPTION
**Pull request by LekoArts**
## Description

This removes the possibility to use `deleteNode` with the `ID` of the node. This was deprecated in favor of passing in the whole `node` and now will be removed.

### New deprecation

We align `createNode` and `deleteNode` and add a new deprecation notice to deleteNode.
So that at the end it will be `createNode(node)` and `deleteNode(node)` instead of `deleteNode({ node })`

### Documentation

The docs (source plugin guide) are updated.

## Related Issues

[ch23642]

### CODE REVIEW

**Comment by wardpeet** [comment]
> Should we align createNode and deleteNode and add a new deprecation notice to deleteNode?
So that at the end it will be createNode(node) and deleteNode(node) instead of deleteNode({ node })

@LekoArts I would say yes

**Comment by LekoArts** [comment]
@TylerBarnes To get myself unblocked and you not urged to merge your tests ASAP I reverted the WordPress changes :) The current state is now using deprecated APIs then, but it still works.

File: packages/gatsby/src/db/__tests__/nodes.js [hunk 1]
```diff
@@ -336,14 +324,17 @@ describe(`nodes db tests`, () => {
...
+        {
+          name: `tests`,
+        }
+      )
     )
     expect(getNode(`hi`)).toBeUndefined()
     const deprecationNotice =
-      `Calling "deleteNode" with a nodeId is deprecated. Please pass an ` +
-      `object containing a full node instead: deleteNode({ node }). ` +
+      `Calling "deleteNode" with an object containing a full node is deprecated. Please pass ` +
```
Comment (laurieontech) [inline packages/gatsby/src/db/__tests__/nodes.js:336]: I might change this slightly?

"Calling 'deleteNode' with {node} is deprecated. Please pass the node directly to the function: deleteNode(node)."

An object containing the full node reads a bit confusing in my mind.

File: packages/gatsby/src/redux/actions/public.js [hunk 2]
```diff
@@ -425,53 +425,50 @@ ${reservedFields.map(f => `  * "${f}"`).join(`\n`)}
...
-  if (typeof options === `string`) {
+  // TODO(v4): Remove this deprecation warning and only allow deleteNode(node)
+  if (node && node.node) {
     let msg =
-      `Calling "deleteNode" with a nodeId is deprecated. Please pass an ` +
-      `object containing a full node instead: deleteNode({ node }).`
-    if (args && args.name) {
-      // `plugin` used to be the third argument
-      plugin = args
+      `Calling "deleteNode" with an object containing a full node is deprecated. Please pass ` +
```
Comment (laurieontech) [inline packages/gatsby/src/redux/actions/public.js:438]: See above.

**Review by laurieontech — COMMENTED** [review]
Left a minor comment on the error message, but it's non-blocking and just a suggestion.

I did notice that this typescript definition isn't updated as part of the PR.   https://github.com/gatsbyjs/gatsby/blob/e691342e5f62dc4156b4e501edf260c773203c8b/packages/gatsby/index.d.ts#L1130

I assume that's generated?

**Comment by LekoArts** [comment]
Nope, I forgot the TS definition. Currently not autogenerated.

**Comment by LekoArts** [comment]
It is updated? :) 

https://github.com/gatsbyjs/gatsby/pull/29205/files#diff-15724d4352562f842e4eb9df3a864711b1617ea4ebabc56c2aee4e975bca6767

**Comment by laurieontech** [comment]
No idea how I missed that 😅

**Pull request merged**